contracts/EscrowAndStaking/CrossChainCollector.sol

```diff
@@ -0,0 +1,407 @@
+            receiver: abi.encode(collectorAddress[_chainSelector]), // ABI-encoded receiver address
+            data: _message, // ABI-encoded string
+            tokenAmounts: new Client.EVMTokenAmount[](0), // Empty array indicating no tokens are being sent
+            extraArgs: Client._argsToBytes(
+                // Additional arguments, setting gas limit and non-strict sequencing mode
+                Client.EVMExtraArgsV1({gasLimit: gasLimitUsed, strict: false})
+            ),
+            // Set the feeToken  address, indicating LINK will be used for fees
+            feeToken: address(0)
+        });
+
+        // Get the fee required to send the message
+        uint256 fees = s_router.getFee(_chainSelector, evm2AnyMessage);
+
+        if (fees > address(this).balance) revert NotEnoughBalance(address(this).balance, fees);
+
+        // Send the message through the router and store the returned message ID
+        messageId = s_router.ccipSend{value: fees}(_chainSelector, evm2AnyMessage);
+
+        // Emit an event with message details
+        emit MessageSent(messageId, _chainSelector, collectorAddress[_chainSelector], _message, address(0), fees);
+
+        // Return the message ID
+        return messageId;
+    }
+
+    function _updateRewardsOnStakingContract(bytes memory data) internal {
+        (uint baseRewards, uint extraRewards, uint revShare) = abi.decode(data, (uint, uint, uint));
+        _updateRewards(baseRewards, extraRewards, revShare);
+    }
+
+    function _updateRewards(
+        uint _baseRewards,
+        uint _extraRewards,
+        uint _revShare
+    ) internal {
+        IStakingThales(stakingThales).updateStakingRewards(_baseRewards, _extraRewards, _revShare);
+    }
+
+    function _setRewardsForNextPeriod() internal {
+        baseRewardsPerPeriod = (baseRewardsPerPeriod * weeklyRewardsDecreaseFactor) / 1e18;
+        extraRewardsPerPeriod = (extraRewardsPerPeriod * weeklyRewardsDecreaseFactor) / 1e18;
+    }
+
+    /* ========== CONTRACT SETTERS FUNCTIONS ========== */
+
+    function setStakingThales(address _stakingThales) external onlyOwner {
+        stakingThales = _stakingThales;
+        emit SetStakingThales(_stakingThales);
+    }
+
+    function setCCIPRouter(address _router) external onlyOwner {
+        _setRouter(_router);
+        s_router = IRouterClient(_router);
+        emit SetCCIPRouter(_router);
+    }
+
+    function setPeriodRewards(
+        uint _baseRewardsPerPeriod,
+        uint _extraRewardsPerPeriod,
+        uint _weeklyDecreaseFactor
+    ) external onlyOwner {
+        baseRewardsPerPeriod = _baseRewardsPerPeriod;
+        extraRewardsPerPeriod = _extraRewardsPerPeriod;
+        weeklyRewardsDecreaseFactor = _weeklyDecreaseFactor;
+        emit SetPeriodRewards(_baseRewardsPerPeriod, _extraRewardsPerPeriod, _weeklyDecreaseFactor);
+    }
+
+    function setGasLimit(uint _gasLimitUsed) external onlyOwner {
```

Comment:
use require, like not above some MAX_GAS

also to not comment all `onlyOwner` functions keep in mind some require statements, which will decrease human error

Reply:
agreed, I will place a MAX_GAS of 10 milion